pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java

```diff
@@ -72,68 +67,27 @@
    * @param config The controller configuration object
    */
   public SegmentStatusChecker(PinotHelixResourceManager pinotHelixResourceManager, ControllerConf config, ControllerMetrics metricsRegistry) {
+    super("SegmentStatusChecker", config.getStatusCheckerFrequencyInSeconds());
     _pinotHelixResourceManager = pinotHelixResourceManager;
     _helixAdmin = pinotHelixResourceManager.getHelixAdmin();
     _config = config;
-    _segmentStatusIntervalSeconds = config.getStatusCheckerFrequencyInSeconds();
     _waitForPushTimeSeconds = config.getStatusCheckerWaitForPushTimeInSeconds();
     _metricsRegistry = metricsRegistry;
     HttpConnectionManager httpConnectionManager = new MultiThreadedHttpConnectionManager();
+  }
 
-    _executorService = Executors.newSingleThreadScheduledExecutor(new ThreadFactory() {
-      @Override
-      public Thread newThread(Runnable runnable) {
-        Thread thread = new Thread(runnable);
-        thread.setName("Segment status checker");
-        return thread;
-      }
-    });
+  @Override
+  public void run() {
+    updateSegmentMetrics();
   }
 
-  /**
-   * Starts the segment status checker.
-   */
-  public void start() {
-    if (_segmentStatusIntervalSeconds == -1) {
+  @Override
+  public void init() {
```

Comment:
Remove this because it's already checked?

Reply:
Yeah sure. The check can be removed while `setStatusToDefault()` needs to be called here.

Reply:
The check has been removed in the last push.